site/docs/v1/tech/lambdas/testing-lambdas.md

```diff
@@ -0,0 +1,380 @@
+```
+
+This test function uses `fetchMock` to mock the external service that would be called from the lambda function in FusionAuth. The mocks for the JWT, user, and registration objects are all simple `{}` objects you can pass as parameters to the `populate()` lambda.
+
+Finally, run the tests.
+
+```bash
+node test.js  | npx faucet
+```
+
+If all your unit tests for a lambda pass, you can safely upload it to FusionAuth manually or with the CLI for further testing.
+
+If your HTTP Connect fetch request fails when deployed to FusionAuth, please review the [documentation](/docs/v1/tech/lambdas/#using-lambda-http-connect). In particular, ensure you are using a license and have purchased the correct plan (Essentials or Enterprise).
+
+### Unit Test: Populate JWT From FusionAuth
+
+In this final unit test, let's look at how to check user information available in FusionAuth to determine custom fields to return to your app.
+
+There are two objects to consider. The first is the JWT fields that are returned to your app by default when a user logs in.
+
+```js
+{
+  aud: 'e9fdb985-9173-4e01-9d73-ac2d60d1dc8e',
+  exp: 1692273965,
+  iat: 1692270365,
+  iss: 'acme.com',
+  sub: '00000000-0000-0000-0000-111111111111',
+  jti: '47cc65ce-3981-4e40-8c99-5221d69c53da',
+  authenticationType: 'PASSWORD',
+  email: 'richard@example.com',
+  email_verified: true,
+  applicationId: 'e9fdb985-9173-4e01-9d73-ac2d60d1dc8e',
+  roles: [ 'Petty cash keeper', 'First aider' ],
+  auth_time: 1692270365,
+  tid: 'd7d09513-a3f5-401c-9685-34ab6c552453',
+  message: 'goodbye world'
+}
+```
+
+The second object is the user supplied to your `populate()` function in a lambda.
+
+```js
+user: {
+    active: true,
+    birthDate: '1985-11-23',
+    connectorId: 'e3306678-a53a-4964-9040-1c96f36dda72',
+    data: {},
+    email: 'richard@example.com',
+    firstName: 'Richard',
+    id: '00000000-0000-0000-0000-111111111111',
+    insertInstant: 1692176681681,
+    lastLoginInstant: 1692270415045,
+    lastName: 'Hendricks',
+    lastUpdateInstant: 1692176681681,
+    memberships: [],
+    passwordChangeRequired: false,
+    passwordLastUpdateInstant: 1692176681694,
+    preferredLanguages: [],
+    registrations: [
+      {
+        applicationId: 'e9fdb985-9173-4e01-9d73-ac2d60d1dc8e',
+        data: {},
+        id: '631b3ef1-4f46-42d2-b315-8d29c1cf18ff',
+        insertInstant: 1692176681698,
+        lastLoginInstant: 1692270415045,
+        lastUpdateInstant: 1692269815209,
+        preferredLanguages: [],
+        roles: [ 'Petty cash keeper', 'First aider' ],
+        tokens: {},
+        usernameStatus: 'ACTIVE',
+        verified: true
+      }
+    ],
+    tenantId: 'd7d09513-a3f5-401c-9685-34ab6c552453',
+    twoFactor: { methods: [], recoveryCodes: [] },
+    usernameStatus: 'ACTIVE',
+    verified: true
+  }
+```
+
+You can see that the user object has data that the JWT does not, like names, birthdates, and languages, that you might want to add in a lambda.
```

Comment:
I'm sorry I was vague here. What I'd love to do is see someone able to:

* download their lambda via the node CLI
* run a unit test against that lambda where the test code is outside of the directory, but the downloaded lambda is being tested
* have the user and other objects mocked up by the test

Does that make sense? What you provided is helpful, but the user will still have to copy and paste their lambda code into `test.js` every time it is downloaded.

Can we import the lambda function into the test context?